packages/app-desktop/gui/MainScreen/commands/perNotebookSortOrder.ts

```diff
@@ -0,0 +1,212 @@
+	title: false,
+	order: false,
+};
+
+export function selectedNotebookHasOwnSortOrder() {
+	return hasOwnSortOrder(getSelectedNotebookId());
+}
+
+export function hasOwnSortOrder(notebookId: string) {
+	return notebookId && ownSortOrders && ownSortOrders.hasOwnProperty(notebookId + SUFFIX_FIELD);
+}
+
+export const declaration: CommandDeclaration = {
+	name: PER_NOTEBOOK_SORT_ORDER,
+	label: () => _('Toggle own sort order'),
+};
+
+export const runtime = (): CommandRuntime => {
+	loadOwnSortOrders();
+	loadSharedSortOrder();
+	eventManager.appStateOn('notesParentType', onNotebookSelectionMayChange);
+	eventManager.appStateOn('selectedFolderId', onNotebookSelectionMayChange);
+
+	return {
+		enabledCondition: 'oneFolderSelected',
+
+		execute: async (context: CommandContext, notebookId?: string, own?: boolean) => {
+			let targetId = notebookId;
+			const selectedId = getSelectedNotebookId(context.state);
+			if (!targetId) {
+				targetId = selectedId; // default: selected notebook
+				if (!targetId) return;
+			}
+			const targetOwn = hasOwnSortOrder(targetId);
+			let newOwn;
+			if (typeof own == 'undefined') {
+				newOwn = !targetOwn; // default: toggling
+			} else {
+				newOwn = !!own;
+				if (newOwn === targetOwn) return;
+			}
+			if (newOwn) {
+				let field: string, reverse: boolean;
+				if (!hasOwnSortOrder(selectedId)) {
+					field = getCurrentField();
+					reverse = getCurrentReverse();
+				} else {
+					field = sharedSortOrder.field;
+					if (perFieldReversalEnabled()) {
+						reverse = sharedSortOrder[field];
+					} else {
+						reverse = sharedSortOrder.reverse;
+					}
+				}
+				setOwnSortOrder(targetId, field, reverse);
+			} else {
+				deleteOwnSortOrder(targetId);
+			}
+		},
+	};
+};
+
+function onNotebookSelectionMayChange() {
+	const selectedId = getSelectedNotebookId();
+	if (previousNotebookId === null) previousNotebookId = selectedId;
+	if (previousNotebookId === selectedId) return;
+	const field = getCurrentField();
+	const reverse = getCurrentReverse();
+	const previousNotebookHasOwnSortOrder = hasOwnSortOrder(previousNotebookId);
+	if (previousNotebookHasOwnSortOrder) {
+		setOwnSortOrder(previousNotebookId, field, reverse);
+	} else {
+		setSharedSortOrder(field, reverse);
+	}
+	previousNotebookId = selectedId;
+	let next;
+	if (hasOwnSortOrder(selectedId)) {
+		next = getOwnSortOrder(selectedId);
+	} else if (previousNotebookHasOwnSortOrder) {
+		next = sharedSortOrder;
+	} else {
+		return;
+	}
+	if (perNotebookSortOrderEnabled()) {
+		if (next.field != field || next.reverse != reverse) {
+			void CommandService.instance().execute(NOTES_SORT_ORDER_SWITCH, next.field, next.reverse);
+		}
+	}
+}
+
+const OWN_SORT_ORDERS = 'notes.perNotebookSortOrders';
+const SHARED_SORT_ORDER = 'notes.sharedSortOrder';
+const SUFFIX_FIELD = '$field';
+const SUFFIX_REVERSE = '$reverse';
+
+function getSelectedNotebookId(state?: State): string {
+	const s = state ? state : app().store().getState();
```

Comment:
Ideally we should never call getState() directly - instead you should only pass the state to the function.

Reply:
finished. 5d0b0e3